### PR TITLE
add acme-dns plugin

### PR DESCRIPTION
**Pull request by webner**
I added a simple dsn plugin for acme dns

### CODE REVIEW

**Comment by Neilpang** [comment]
check this guide first: https://github.com/Neilpang/acme.sh/wiki/DNS-API-Dev-Guide

File: dnsapi/dns_acmedns.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  export _H1="X-Api-User: $ACMEDNS_USERNAME"
+  export _H2="X-Api-Key: $ACMEDNS_PASSWORD"
+  data="{\"subdomain\":\"$ACMEDNS_SUBDOMAIN\", \"txt\": \"$txtvalue\"}"
+
+  _debug data "$data"
+  response="$(_post "$data" "$ACMEDNS_UPDATE_URL" "" "POST")"
+  _debug response "$response"
+
+  if ! printf "%s" "$response" | grep "\"$txtvalue\"" >/dev/null; then
```
Comment (Neilpang) [inline dnsapi/dns_acmedns.sh]: please use `echo` instead.

File: dnsapi/dns_acmedns.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    _err "invalid response of acme-dns"
+    return 1
+  fi
+
+}
+
+#Usage: fulldomain txtvalue
+#Remove the txt record after validation.
+dns_acmedns_rm() {
+  fulldomain=$1
```
Comment (Neilpang) [inline dnsapi/dns_acmedns.sh]: the rm() function is not needed ?

Comment (webner) [reply dnsapi/dns_acmedns.sh]: acme-dns has no remove method

Comment (Neilpang) [reply dnsapi/dns_acmedns.sh]: I wonder is it needed to remove the record ?

Comment (webner) [reply dnsapi/dns_acmedns.sh]: As far as I understand it acme-dns can only hold two txt values for one record and the old ones are overwritten. So no I don't think it is needed.

**Comment by Neilpang** [comment]
add to the main readme dns api list.

**Comment by Neilpang** [comment]
please merge the latest code

**Comment by Neilpang** [comment]
please merge the latest code

**Comment by webner** [comment]
@Neilpang Is there anything else I should do?

**Pull request merged**